File: src/app/view/KeyAlert.java
package app.view;

import app.App;
import app.Config;
import javafx.scene.control.Alert;
import javafx.scene.control.ButtonType;

import static javafx.scene.control.ButtonBar.ButtonData.CANCEL_CLOSE;

class KeyAlert extends Alert {
    static ButtonType GENERATE;
    static ButtonType CHOOSE;
    private static ButtonType CANCEL;

    static {
        GENERATE = new ButtonType("Generate key files");
        CHOOSE = new ButtonType("Choose a key file");
        CANCEL = new ButtonType("Cancel", CANCEL_CLOSE);
    }

    KeyAlert() {
        super(AlertType.CONFIRMATION);
        this.setTitle("Choose an action");
        this.getButtonTypes().setAll(CANCEL, CHOOSE, GENERATE);

        App.getRsa().ifPresentOrElse(
                rsa -> {
                    this.setHeaderText("Current RSA Key file is: " + Config.getLastRSAKeyPath());
                    this.setContentText("Modulus: " + rsa.getKey().getModulus() + "\nPower: " + rsa.getKey().getPower());
                },
                () -> this.setHeaderText("RSA Key is not set.")
        );
    }
}

File: src/app/view/KeyAlert.kt
package app.view

import app.App
import app.Config
import javafx.scene.control.Alert
import javafx.scene.control.ButtonBar.ButtonData.CANCEL_CLOSE
import javafx.scene.control.ButtonType

internal class KeyAlert : Alert(Alert.AlertType.CONFIRMATION) {
    init {
        this.title = "Choose an action"
        this.buttonTypes.setAll(CANCEL, CHOOSE, GENERATE)

        App.getRsa().ifPresentOrElse(
                { rsa ->
                    this.headerText = "Current RSA Key file is: " + Config.lastRSAKeyPath
                    this.contentText = "Modulus: " + rsa.key.modulus + "\nPower: " + rsa.key.power
                }
        ) { this.headerText = "RSA Key is not set." }
    }

    companion object {
        var GENERATE: ButtonType
        var CHOOSE: ButtonType
        private var CANCEL: ButtonType? = null

        init {
            GENERATE = ButtonType("Generate key files")
            CHOOSE = ButtonType("Choose a key file")
            CANCEL = ButtonType("Cancel", CANCEL_CLOSE)
        }
    }
}
